x/smart/stream_dialer.go

```diff
@@ -365,14 +394,58 @@ func (f *StrategyFinder) newProxylessDialer(ctx context.Context, testDomains []s
 	return f.findTLS(ctx, testDomains, dnsDialer, config.TLS)
 }
 
+func (f *StrategyFinder) parseConfig(configBytes []byte) (configConfig, error) {
```

Comment:
Why are you doing this manual parsing instead of unmarshalling into the output directly?

Reply:
I got rid of the manual parsing and switched this to use `mapToAny` + `Unmarshal`.

I've just copied `mapToAny` in wholesale from outline-apps. I figure we'll want it anyway eventually. But let me know if you'd like me to put it somewhere other than this file in anticipation of eventually moving over more code.

I've also added a bit of unit testing, partially just to enumerate the allowed config fields in a single readable place besides `x/examples/smart_proxy`.